cloudflare_ddns

```diff
@@ -45,11 +51,19 @@ list_dns_records_onthrottled()
 
 update_record()
 {
-	OUTPUT=$(curl -s -X PUT "https://api.cloudflare.com/client/v4/zones/$DNS_ZONE_ID/dns_records/$DNS_RECORD_ID" \
-		-H "X-Auth-Email: $CLOUDFLARE_EMAIL" \
-		-H "X-Auth-Key: $CLOUDFLARE_API_KEY" \
+	if [ -z "$CLOUDFLARE_API_TOKEN" ]
+	then
+	        OUTPUT=$(curl -s -X PUT "https://api.cloudflare.com/client/v4/zones/$DNS_ZONE_ID/dns_records/$DNS_RECORD_ID" \
+                -H "X-Auth-Email: $CLOUDFLARE_EMAIL" \
+                -H "X-Auth-Key: $CLOUDFLARE_API_KEY" \
+                -H "Content-Type: application/json" \
+                --data "{\"type\":\"$DNS_RECORD_TYPE\",\"name\":\"$DNS_RECORD_NAME\",\"content\":\"$NEW_IP\",\"proxied\":$PROXY_STATE}")
+	else
+	        OUTPUT=$(curl -s -X PUT "https://api.cloudflare.com/client/v4/zones/$DNS_ZONE_ID/dns_records/$DNS_RECORD_ID" \
```

Comment:
This is quite a bit of duplication. All that's really changing are the authentication headers being sent alongside otherwise identical requests. If you can address this, that would be great.

Reply:
you're right.  Initially I created one condition up top to check for the existence of the new Token and set an AUTH_HEADER variable appropriately.  However, when I tried to insert $AUTH_HEADER into your OUTPUT= statements I was unable to get them to properly make the API calls.  I think it has to do with how the shell expands the quotes around them.  I played with this for well over an hour trying to escape them different ways, but it would never format properly for the JSON request.  In the end I went for function over form as I could get it to work this way.  I'm more than happy for you to streamline it if you can figure it out.

Reply:
I'm a bit overloaded at the moment, but I'll have a look at what's possible this week.

Reply:
@bengalih I've refactored the script a bit to make it easier to incorporate your changes.